mne/decoding/transformer.py

```diff
@@ -520,3 +520,89 @@ def transform(self, epochs_data, y=None):
                                      picks=self.picks, n_jobs=self.n_jobs,
                                      copy=False, verbose=False)
         return epochs_data
+
+
+class UnsupervisedSpatialFilter(TransformerMixin):
+    """Fit and transform with an unsupervised spatial filtering across time
+    and samples.
+
+    Parameters
+    ----------
+    estimator : scikit-learn estimator
+        Estimator using some decomposition algorithm.
+    average : bool, default False
+        If true, average of X is used in estimator's fit.
+    """
+    def __init__(self, estimator, average=False):
+        for attr in ('fit', 'transform', 'fit_transform'):
+            if not hasattr(estimator, attr):
+                raise ValueError('estimator must be a sklearn transformer')
```

Comment:
sklearn -> scikit-learn

then also say what is missing to be a scikit-learn valid transformer